cdap-proto/src/main/java/co/cask/cdap/proto/Id.java

```diff
@@ -383,18 +383,17 @@ public int hashCode() {
    * Id.Stream uniquely identifies a stream.
    */
   public static final class Stream {
-    private final String namespace;
+    private final Namespace namespace;
```

Comment:
We cache the hashcode for StreamId, so the hashcode will only be computed once for the object.

That should be fine, right